backend/config.go

```diff
@@ -122,3 +123,26 @@ func (c *GrafanaCfg) AppURL() (string, error) {
 	}
 	return url, nil
 }
+
+type userAgentKey struct{}
+
+// UserAgentFromContext returns user agent from context.
+func UserAgentFromContext(ctx context.Context) (*useragent.UserAgent, error) {
+	v := ctx.Value(userAgentKey{})
+	if v == nil {
+		return useragent.Empty()
+	}
+
+	ua := v.(*useragent.UserAgent)
+	if ua == nil {
+		return useragent.Empty()
```

Comment:
shouldn't this return an error? In theory this happens when the userAgentKey is set but it's an unrecognized type.

Reply:
This mimics the behaviour of `GrafanaConfigFromContext` [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/config.go#L28), but of course I am happy to change it if you'd prefer. :)

Reply:
Definitely keep the check still, but is okay to just return empty 👍